File: ios/MessageModule.m
//
//  MessageModule.m
//  NativeBridge
//
//  Created by Hayk Kyutneryan on 06.04.25.
//

#import <Foundation/Foundation.h>

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MessageModule, NSObject)

RCT_EXTERN_METHOD(sendMessage:(NSString *)message
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/MessageModule.swift
//
//  MessageModule.swift
//  NativeBridge
//
//  Created by Hayk Kyutneryan on 06.04.25.
//

import Foundation

@objc(MessageModule)
class MessageModule: NSObject {
  
  @objc
  func sendMessage(_ message: String, resolver resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {
    let response = "Received: \(message)"
    resolve(response)
  }
}
